packages/catalog-model/src/validation/ajv.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Ajv, { Schema, ValidateFunction } from 'ajv';
+import entitySchema from '../schema/Entity.schema.json';
+import entityEnvelopeSchema from '../schema/EntityEnvelope.schema.json';
+import entityMetaSchema from '../schema/EntityMeta.schema.json';
+import commonSchema from '../schema/shared/common.schema.json';
+
+// A local cache of compiled schemas, to avoid duplicate work
+const compiledSchemaCache = new Map<unknown, ValidateFunction<unknown>>();
```

Comment:
Where does this actually end up being used?

Reply:
Myeah could be argued that it's premature and usually isn't a problem because a sensible caller might think this is an expensive operation? I wanted to make it easy to just call this over and over without risk

Reply:
Changed to stringified key